subsys/net/l2/canbus/6locan.c

```diff
@@ -145,22 +145,17 @@ static void canbus_st_min_timeout(struct _timeout *t)
 	k_work_submit_to_queue(&net_canbus_workq, &ctx->pkt->work);
 }
 
-static s32_t canbus_stmin_to_ticks(u8_t stmin)
+static k_timeout_t canbus_stmin_to_ticks(u8_t stmin)
 {
-	s32_t time_ms;
-
 	/* According to ISO 15765-2 stmin should be 127ms if value is corrupt */
 	if (stmin > NET_CAN_STMIN_MAX ||
 	    (stmin > NET_CAN_STMIN_MS_MAX && stmin < NET_CAN_STMIN_US_BEGIN)) {
-		time_ms = K_MSEC(NET_CAN_STMIN_MS_MAX);
+		return K_MSEC(NET_CAN_STMIN_MS_MAX);
 	} else if (stmin >= NET_CAN_STMIN_US_BEGIN) {
-		/* This should be 100us-900us but zephyr can't handle that */
```

Comment:
But now we can!  (Though 100us is in fact the default tick period on most non-nRF hardware platforms now, so this is still pushing the limits of what you can expect to work).

Reply:
That's right!